Description: walinuxagent.service: adjust Wants and After
 There is no reason for walinuxagent.service to Want
 cloud-final.service.  That could cause problems if cloud-init was
 disabled.
 .
 There is also no reason for it to need to run After cloud-final.
 Instead replace that with After of cloud-init.  That will ensure
 that cloud-init in its datasource code has an opportunity affect
 walinuxagent before it starts.
 .
 The need for this is because cloud-init was starting walinux.service
 during cloud-init.service, which would cause unresolvable
 cyclic dependencies.
Author: Scott Moser <smoser@ubuntu.com>
Forwarded: https://github.com/Azure/WALinuxAgent/pull/426
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1623570
Bug-Upstream: https://github.com/Azure/WALinuxAgent/issues/425
Last-Update: 2016-09-14


--- a/init/ubuntu/walinuxagent.service
+++ b/init/ubuntu/walinuxagent.service
@@ -7,8 +7,8 @@
 [Unit]
 Description=Azure Linux Agent
 
-After=network-online.target cloud-final.service
-Wants=network-online.target sshd.service sshd-keygen.service cloud-final.service
+After=network-online.target cloud-init.service
+Wants=network-online.target sshd.service sshd-keygen.service
 
 ConditionFileIsExecutable=/usr/sbin/waagent
 ConditionPathExists=/etc/waagent.conf
